### PR TITLE
CSPL 1180 - CRD apiVersioning change from V2 to V3

### PR DESCRIPTION
**Pull request by kashok-splunk**
CRD apiVersioning change from V2 to V3

### CODE REVIEW

File: pkg/splunk/enterprise/licensemaster.go [hunk 1]
```diff
@@ -133,13 +133,6 @@ func ApplyLicenseMaster(client splcommon.ControllerClient, cr *enterpriseApi.Lic
 		} else {
 			result.Requeue = false
 		}
-
```
Comment (kashok-splunk) [inline pkg/splunk/enterprise/licensemaster.go:136]: Duplicate Code. Was added by mistake when resolving conflicts

File: pkg/splunk/enterprise/searchheadcluster.go [hunk 2]
```diff
@@ -182,13 +182,6 @@ func ApplySearchHeadCluster(client splcommon.ControllerClient, cr *enterpriseApi
 			result.Requeue = false
 		}
 
-		// Requeue the reconcile after polling interval if we had set the lastAppInfoCheckTime.
```
Comment (kashok-splunk) [inline pkg/splunk/enterprise/searchheadcluster.go:185]: Duplicate Code. Was added by mistake when resolving conflicts

File: pkg/splunk/enterprise/standalone.go [hunk 3]
```diff
@@ -195,13 +195,6 @@ func ApplyStandalone(client splcommon.ControllerClient, cr *enterpriseApi.Standa
 		} else {
 			result.Requeue = false
 		}
-
```
Comment (kashok-splunk) [inline pkg/splunk/enterprise/standalone.go:198]: Duplicate Code. Was added by mistake when resolving conflicts

**Review by jryb — APPROVED** [review]

File: build/make_bundle.sh [hunk 4]
```diff
@@ -123,7 +123,7 @@ cat << EOF >$YAML_SCRIPT_FILE
       }
     },
     {
-      "apiVersion": "enterprise.splunk.com/v2",
+      "apiVersion": "enterprise.splunk.com/v3",
       "kind": "Standalone",
       "metadata": {
         "name": "example",
```
Comment (sgontla) [inline build/make_bundle.sh:129]: Do we need to add similar entry for Monitoring Console?

**Review by sgontla — APPROVED** [review]

File: pkg/splunk/enterprise/monitoringconsole.go [hunk 5]
```diff
@@ -127,7 +127,6 @@ func ApplyMonitoringConsole(client splcommon.ControllerClient, cr *enterpriseApi
 		} else {
 			result.Requeue = false
 		}
-		result.Requeue = false
```
Comment (smohan-splunk) [inline pkg/splunk/enterprise/monitoringconsole.go:130]: Just confirming, is this an expected change?

**Review by smohan-splunk — APPROVED** [review]

File: pkg/splunk/enterprise/monitoringconsole.go (hunk 5 above)
Comment (kashok-splunk) [reply pkg/splunk/enterprise/monitoringconsole.go:130]: yes, this is expected. Condition is fulfilled in else block line #128

File: deploy/crds/enterprise.splunk.com_monitoringconsoles_crd.yaml [hunk 6]
```diff
@@ -2729,12 +2729,23 @@ spec:
...
+        properties:
+          apiVersion:
+            type: string
   - name: v1
     served: true
     storage: false
     schema:
       openAPIV3Schema:
         type: object
+        x-kubernetes-preserve-unknown-fields: true
```
[the remark this answers is not in the document]
Comment (kashok-splunk) [reply deploy/crds/enterprise.splunk.com_monitoringconsoles_crd.yaml:2748]: This is MC CRD which is not present in the develop branch

[the remark this answers is not in the document]
Comment (gaurav-splunk) [reply deploy/crds/enterprise.splunk.com_monitoringconsoles_crd.yaml:2748]: oh yeah ok my bad, LGTM then!

**Review by gaurav-splunk — APPROVED** [review]

**Pull request merged**